cmd/storage/postgresql/accrual_sys.go

```diff
@@ -0,0 +1,94 @@
+package storage
+
+import (
+	add "github.com/Tanya1515/gophermarket/cmd/additional"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+func (db *PostgreSQL) ProcessAccOrder(order add.Order) (err error) {
+	var userID int
+	tx, err := db.dbConn.Begin()
+	if err != nil {
+		return
+	}
+
+	if (order.Status == "PROCESSED") && (order.Accrual > 0) {
+		row := tx.QueryRow("SELECT user_id FROM orders WHERE id=$1", order.Number)
+		err = row.Scan(&userID)
+		if err != nil {
+			tx.Rollback()
+			return
+		}
+
+		_, err = tx.Exec("UPDATE orders SET accrual=$1 WHERE id=$2", order.Accrual, order.Number)
+		if err != nil {
+			tx.Rollback()
+			return
+		}
+
+		_, err = tx.Exec("UPDATE users SET sum=sum+$1 WHERE id=$2", order.Accrual, userID)
+		if err != nil {
+			tx.Rollback()
+			return
+		}
+
+	}
+
+	_, err = tx.Exec("UPDATE orders SET status=$1 WHERE id=$2", order.Status, order.Number)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	return
+}
+
+func (db *PostgreSQL) StartProcessingUserOrder(logger zap.SugaredLogger, result chan add.OrderAcc) {
+
+	var order add.OrderAcc
+	g := new(errgroup.Group)
+
+	for {
+		rows, err := db.dbConn.Query("SELECT id, accrual FROM orders WHERE status=$1", "NEW")
```

Comment:
```suggestion
		rows, err := db.dbConn.Query("SELECT id, accrual FROM orders WHERE status IN ($1, $2)", "NEW", "PROCESSING")
```
эти записи тоже нужно захватывать, раз этот статус ловим и сохраняем. Иначе мы их никогда не пересчитаем. А лучше еще и LIMIT 1000 добавить. Могут быть проблемы, если сервис accrual приляжет на пару дней, то в БД скопится много записей, которые довольно тяжко будет вытаскивать каждый раз.